Fix: Bug where save profile as <name>.xml would save as <name>.xml.xml

odavison opened the pull request:
The functions responsible for exporting profiles and triggers were
blindly appending ".xml" onto the end of the filename before writing.
This would result in exporting a file called <name>.xml.xml if the
user specified <name>.xml. This commit adds a case-insensitive check
to see if the filename already ends in ".xml" before appending it
to the end of the filename.

zcdziura commented:
Thank you for fixing this, it's always been a pain having to manually remove that extra ".xml" extension from the file name!

Chris7 commented on src/dlgTriggerEditor.cpp:
```diff
@@ -6993,7 +6993,10 @@ void dlgTriggerEditor::slot_profileSaveAsAction()
 
     if(fileName.isEmpty()) return;
-    fileName.append(".xml");
+    if (!fileName.endsWith(".xml"))
```
You probably want to cast filename to lower case first to handle anyone who decides to capitalize xml

odavison commented:
I hadn't considered users inputting.XML (or variants). Thanks for pointing that out!

vadi2 commented:
Looks like your contribution is fine, just one OSX build is now broken for some reason.

SlySven commented:
:hammer:  After reviewing the log for that failed Travis C.I. check, I kicked it to retry that one build  that unexpectedly failed and it now passes - it was a glitch on setting up the build environment (some libraries, including Qt5 failed to install!) but it seems all right now... :grinning:

odavison commented:
Thanks for giving it the 'kick'. It's handy to know that's possible without pushing another commit.

odavison commented:
Sincere apologies it took me this long to make this small fix. I've updated the branch per your comments re: QStringLiteral and Qt::CaseInsensitive, added a fix for the trigger export that had the same problem, squashed commits, and rebased onto development.
